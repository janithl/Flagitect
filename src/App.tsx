import React from 'react';

import {
  AppState,
  Editor,
  EditorHeader,
  EditorModal,
  Footer,
} from '@components';

export default (): JSX.Element => (
  <AppState>
    <EditorHeader />
    <Editor />
    <Footer />
    <EditorModal />
  </AppState>
);
