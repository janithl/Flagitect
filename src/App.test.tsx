import React from 'react';
import { render } from '@testing-library/react-native';

import App from './App';

test('Renders initial app state', () => {
  const { getByText, getByTestId } = render(<App />);

  expect(getByText('Division')).toBeTruthy();
  expect(getByText('Vertical')).toBeTruthy();

  expect(getByText('Proportion')).toBeTruthy();
  expect(getByText('2 : 3')).toBeTruthy();

  expect(getByText('Colours')).toBeTruthy();

  expect(getByTestId('editor').children).toMatchSnapshot();
});
