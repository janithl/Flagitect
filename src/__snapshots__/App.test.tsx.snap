// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders initial app state 1`] = `
Array [
  <RNSVGSvgView
    bbHeight={480}
    bbWidth={720}
    fill="#FFFFFF"
    focusable={false}
    height={480}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 480,
          "width": 720,
        },
      ]
    }
    width={720}
    xmlns="http://www.w3.org/2000/svg"
  >
    <RNSVGGroup
      fill={4294967295}
      propList={
        Array [
          "fill",
        ]
      }
    >
      <RNSVGGroup>
        <RNSVGRect
          fill={4279918516}
          height={480}
          propList={
            Array [
              "fill",
            ]
          }
          width={240}
          x={0}
          y="0"
        />
        <RNSVGRect
          fill={4294967295}
          height={480}
          propList={
            Array [
              "fill",
            ]
          }
          width={240}
          x={240}
          y="0"
        />
        <RNSVGRect
          fill={4289010989}
          height={480}
          propList={
            Array [
              "fill",
            ]
          }
          width={240}
          x={480}
          y="0"
        />
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>,
]
`;
